libcudacxx/include/cuda/std/detail/libcxx/include/version

```diff
@@ -201,234 +201,464 @@ __cpp_lib_void_t                                        201411L <type_traits>
 
 #ifndef __cuda_std__
 #include <__config>
-#include <__pragma_push>
+#endif // __cuda_std__
+
+// If we are replacing the stl we need to define our own feature test macros.
+// However, if we are in __cuda_std__ we need to define our own symbols to not
+// conflict with the host stl.
+// At the same time we want bring in all feature test macros from host
+#ifdef __cuda_std__
+#if __has_include(<version>) // <version> should be the smallest include possible
+#include <version>
+#elif !defined(_LIBCUDACXX_COMPILER_NVRTC)
+#include <ciso646> // otherwise go for the smallest possible header
+#endif
+#endif // __cuda_std__
 
 #if defined(_LIBCUDACXX_USE_PRAGMA_GCC_SYSTEM_HEADER)
 #pragma GCC system_header
 #endif
 
+// We unconditionally define `__cccl_lib_meow` so that there is only one place to set the value
 #if _LIBCUDACXX_STD_VER > 11
-# define __cpp_lib_chrono_udls                          201304L
-# define __cpp_lib_complex_udls                         201309L
+# define __cccl_lib_chrono_udls                          201304L
+# define __cccl_lib_complex_udls                         201309L
 #ifdef _LIBCUDACXX_IS_CONSTANT_EVALUATED
-# define __cpp_lib_constexpr_complex                    201711L
+# define __cccl_lib_constexpr_complex                    201711L
 #endif
-# define __cpp_lib_concepts                             202002L
-# define __cpp_lib_exchange_function                    201304L
+# define __cccl_lib_concepts                             202002L
+# define __cccl_lib_exchange_function                    201304L
+// # define __cccl_lib_generic_associative_lookup           201304L
+# define __cccl_lib_integer_sequence                     201304L
+# define __cccl_lib_integral_constant_callable           201304L
+# define __cccl_lib_is_final                             201402L
+# define __cccl_lib_is_null_pointer                      201309L
+# define __cccl_lib_make_reverse_iterator                201402L
+// # define __cccl_lib_make_unique                          201304L
+# define __cccl_lib_null_iterators                       201304L
+// # define __cccl_lib_quoted_string_io                     201304L
+# define __cccl_lib_result_of_sfinae                     201210L
+# define __cccl_lib_robust_nonmodifying_seq_ops          201304L
+# if !defined(_LIBCUDACXX_HAS_NO_THREADS)
+// #   define __cccl_lib_shared_timed_mutex                 201402L
+# endif
+# define __cccl_lib_span                                 201902L
+// # define __cccl_lib_string_udls                          201304L
+# define __cccl_lib_transformation_trait_aliases         201304L
+# define __cccl_lib_transparent_operators                201510L
```

Comment:
It used to be `201210L` for C++11 and `201510L` for C++14+. I'm not sure if the change is safe. If it is, please add a comment explaining why.

Reply:
This comes from That was indeed an error, I removed the updated define for C++17 but somehow managed to change this definition. 

It should be 2012 because we do not implement `shared_ptr` whose comparison objects are what signifies the increase to 2015 (See https://en.cppreference.com/w/cpp/feature_test)